app/controllers/users/sessions_controller.rb

```diff
@@ -204,6 +205,32 @@ def override_csp_for_google_analytics
     def sign_in_params
       params[resource_name]&.permit(:email) if request.post?
     end
+
+    def check_password_compromised
+      return if current_user.password_compromised_checked_at.present? ||
+                !eligible_for_password_lookup?
+
+      session[:redirect_to_password_compromised] =
+        PwnedPasswords::LookupPassword.call(auth_params[:password])
+      update_user_password_compromised_checked_at
+    end
+
+    def eligible_for_password_lookup?
+      FeatureManagement.check_password_enabled? &&
+        randomize_check_password?
+    end
+
+    def update_user_password_compromised_checked_at
+      UpdateUser.new(
+        user: current_user,
+        attributes: { password_compromised_checked_at: Time.zone.now },
+      ).call
+    end
+
+    def randomize_check_password?
+      SecureRandom.random_number(IdentityConfig.store.compromised_password_randomizer_value) >
+        IdentityConfig.store.compromised_password_randomizer_threshold
```

Comment:
The development configuration you added never triggers for me testing in local development, because `SecureRandom.random_number` will generate a number [`0 <= SecureRandom.random_number(n) < n`](https://rubydoc.info/stdlib/securerandom/1.9.3/SecureRandom.random_number), so it'll always be `0`.

I think it'll work if we swap `>` with `>=` (`0 >= 0`), or alternatively change the `compromised_password_randomizer_threshold` development configuration to `-1` (`0 > -1`):

```suggestion
      SecureRandom.random_number(IdentityConfig.store.compromised_password_randomizer_value) >=
        IdentityConfig.store.compromised_password_randomizer_threshold
```